src/main/java/javax/validation/constraints/FutureOrPresent.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Bean Validation API
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package javax.validation.constraints;
+
+import static java.lang.annotation.ElementType.ANNOTATION_TYPE;
+import static java.lang.annotation.ElementType.CONSTRUCTOR;
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.ElementType.TYPE_USE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+import java.time.Year;
+
+import javax.validation.ClockProvider;
+import javax.validation.Constraint;
+import javax.validation.Payload;
+import javax.validation.constraints.FutureOrPresent.List;
+
+/**
+ * The annotated element must be an instant, date or time in the present or in the future.
+ * <p>
+ * <i>Now</i> is defined by the {@link ClockProvider} attached to the {@link javax.validation.Validator}
```

Comment:
I'd prefer simple names (via imports) as opposed to FQNs in JavaDoc. Can be done as a follow-up, though.